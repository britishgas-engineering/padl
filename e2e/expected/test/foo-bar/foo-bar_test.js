suite('foo-bar', function() {

  test('instantiating the element with default properties works', async () => {
    const element = await fixture('fooBarBasicTestFixture');
    const elementShadowRoot = element.shadowRoot;
    const elementHeader = elementShadowRoot.querySelector('h2');

    assert.equal(element.prop1, 'foo-bar', 'prop1 has value `foo-bar`');
    assert.equal(elementHeader.innerHTML, 'Hello foo-bar!', 'header has correct value');
    a11ySuite('fooBarBasicTestFixture');
  });

  test('setting a property on the element works', async () => {
    const element = await fixture('fooBarChangedPropertyTestFixture');
    const elementShadowRoot = element.shadowRoot;
    const elementHeader = elementShadowRoot.querySelector('h2');

    assert.equal(element.prop1, 'new-prop1', 'prop1 has value `new-prop1`');
    assert.equal(elementHeader.innerHTML, 'Hello new-prop1!', 'header has correct value');
    a11ySuite('fooBarChangedPropertyTestFixture');
  });

});
