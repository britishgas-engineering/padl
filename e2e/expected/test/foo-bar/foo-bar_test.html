<test-fixture id="fooBarBasicTestFixture">
  <template>
    <foo-bar></foo-bar>
  </template>
</test-fixture>

<test-fixture id="fooBarChangedPropertyTestFixture">
  <template>
    <foo-bar prop1="new-prop1"></foo-bar>
  </template>
</test-fixture>
